src/main/java/uk/gov/ons/ctp/response/action/service/ActionService.java

```diff
@@ -173,25 +200,129 @@ public Action createAction(final Action action) {
+        throw new IllegalStateException(NO_RESPONDENTS_FOR_SURVEY);
+      }
+
+      enrolledAssociations.forEach(
+          association ->
+              createAction(
+                  actionCase, actionRule, actionType, UUID.fromString(association.getPartyId())));
+
+    } else {
+      createAction(actionCase, actionRule, actionType, actionCase.getPartyId());
+    }
+  }
+
+  private List<Association> respondentsEnrolledOnCase(ActionCase actionCase) {
+    PartyDTO businessParty =
+        partySvcClientService.getParty(actionCase.getSampleUnitType(), actionCase.getPartyId());
+    CollectionExerciseDTO collectionExercise =
+        collectionExerciseClientService.getCollectionExercise(actionCase.getCollectionExerciseId());
+    return associationsEnrolledForSurvey(businessParty, collectionExercise.getSurveyId());
+  }
+
+  private List<Association> associationsEnrolledForSurvey(PartyDTO party, String surveyId) {
+    return party
+        .getAssociations()
         .stream()
-        .map(pa -> Action.fromPotentialAction(pa, currentTime))
-        .forEach(a -> actionRepo.save(a));
-    actionRepo.flush();
+        .filter(association -> isAssociationEnabledForSurvey(association, surveyId))
+        .collect(Collectors.toList());
+  }
+
+  private boolean isAssociationEnabledForSurvey(Association association, String surveyId) {
+    return association
+        .getEnrolments()
+        .stream()
+        .anyMatch(
+            enrolment ->
+                enrolment.getSurveyId().equals(surveyId)
+                    && enrolment.getEnrolmentStatus().equalsIgnoreCase(ENABLED));
+  }
+
+  private void createAction(
+      ActionCase actionCase, ActionRule actionRule, ActionType actionType, UUID partyId) {
+    if (actionRepo.findOneByCaseIdAndActionRuleFKAndPartyId(
+            actionCase.getId(), actionRule.getActionRulePK(), partyId)
+        != null) {
+      log.with("caseId", actionCase.getId().toString())
+          .with("actionRuleId", actionRule.getId().toString())
+          .with("partyId", partyId.toString())
+          .debug("Action already exists");
+      return;
+    }
+
+    Action newAction = new Action();
+    newAction.setId(UUID.randomUUID());
+    newAction.setCreatedBy("SYSTEM");
+    newAction.setManuallyCreated(false);
+    newAction.setState(ActionState.SUBMITTED);
+    newAction.setCreatedDateTime(new Timestamp((new Date()).getTime()));
+
+    newAction.setCaseFK(actionCase.getCasePK());
+    newAction.setCaseId(actionCase.getId());
+
+    newAction.setActionPlanFK(actionRule.getActionPlanFK());
+    newAction.setActionRuleFK(actionRule.getActionRulePK());
+    newAction.setActionType(actionType);
+    newAction.setPriority(actionRule.getPriority());
 
+    newAction.setPartyId(partyId);
+
+    actionRepo.saveAndFlush(newAction);
+  }
+
+  private void updatePlanAndJob(ActionPlan actionPlan, ActionPlanJob actionPlanJob) {
+    final Timestamp currentTime = new Timestamp((new Date()).getTime());
```

Comment:
Time zone safe?

Reply:
I think this should default to UTC which is what we want to be saving in the database?

Reply:
Still... DateTimeUtil.nowUTC() is used elsewhere to get a timestamp. Make it consistent?